### PR TITLE
Delete hanging jobs (DeadlineExceeded, Active == 1 but actually completed)

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
I've found that we have 2 kinds of jobs kept hanging for weeks in our cluster:

1) The ones that exceeded their deadline `DeadlineExceeded`, the problem in this case is that k8s does not set `job.Status.Failed` so the solution is to check that there is a job condition of type `Failed` and state `true`

Real example of a Job JSON stuck in this state:

```json
{
    "status": {
        "conditions": [
            {
                "lastProbeTime": "2020-07-23T17:50:10Z",
                "lastTransitionTime": "2020-07-23T17:50:10Z",
                "message": "Job was active longer than specified deadline",
                "reason": "DeadlineExceeded",
                "status": "True",
                "type": "Failed"
            }
        ],
        "startTime": "2020-07-23T17:40:10Z"
    }
}
```

2) The ones that have finished successfully, but for some reason their `job.Status.Active` is still `1`, even though there are no related pods. I suspect it's a bug in k8s, maybe a race with job completion and pod being deleted by this controller (just a guess). The solution here is to rely on `jobFinishTime` rather than `Status.active`, `jobFinishTime` is non-zero only and only if Job either completed successfully or failed.

Real example of a Job JSON stuck in this state:

```json
{
    "status": {
        "active": 1,
        "completionTime": "2020-06-19T02:43:35Z",
        "conditions": [
            {
                "lastProbeTime": "2020-06-19T02:43:35Z",
                "lastTransitionTime": "2020-06-19T02:43:35Z",
                "status": "True",
                "type": "Complete"
            }
        ],
        "startTime": "2020-06-19T02:17:58Z",
        "succeeded": 1
    }
}
```

---

I've verified the build with these changes with dry-run and then in prod mode in our cluster and it successfully deleted hanging jobs while not touching the running ones.

### CODE REVIEW

**Comment by lwolf** [comment]
thanks for the PR. New day, new edge-case :exploding_head: 
Can you please add a testcase for this

**Comment by artem-zinnatullin** [comment]
Added tests, sorry if my Go programming is a bit off, it's not my daily language :)

File: pkg/controller/job_test.go [hunk 1]
```diff
@@ -39,54 +41,93 @@ func TestKleaner_DeleteJob(t *testing.T) {
...
+					Reason:             "",
+					Message:            "",
+				},
+			}),
+			successful: time.Second,
+			failed:     time.Second,
+			ignoreCron: false,
+			expected:   true,
+		},
+		"failed but 'active' jobs should be deleted": {
```
Comment (lwolf) [inline pkg/controller/job_test.go:124]: shouldn't here be jobConditions? I don't quite understand this case.

Comment (artem-zinnatullin) [reply pkg/controller/job_test.go:124]: I specifically didn't add it here, because if we see `"failed": 1` — it's enough to distinguish the job as failed (I haven't seen failed job with `"active": 1`, that's why)

But you're right, adding `jobConditions` is another test case I can add — will do!

**Comment by artem-zinnatullin** [comment]
Added another test case to cover your very reasonable question, improved test naming to reflect what's going on better

**Comment by lwolf** [comment]
great, thank you!

**Comment by artem-zinnatullin** [comment]
Thanks for the review and a quick release! :)
